build/configuration-properties-collector/src/main/java/org/hibernate/search/configuration/properties/collector/impl/AsciiDocWriter.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.configuration.properties.collector.impl;
+
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class AsciiDocWriter implements BiConsumer<Map<String, ConfigurationProperty>, Writer> {
+
+	private final Optional<String> moduleName;
+	private final Predicate<Map.Entry<String, ConfigurationProperty>> filter;
+
+	public AsciiDocWriter(String moduleName, Predicate<Map.Entry<String, ConfigurationProperty>> filter) {
+		this.moduleName = Optional.ofNullable( moduleName );
+		this.filter = filter;
+	}
+
+	@Override
+	public void accept(Map<String, ConfigurationProperty> propertyMap, Writer writer) {
+		List<Map.Entry<String, ConfigurationProperty>> entries = propertyMap.entrySet().stream()
+				.filter( filter )
+				.collect( Collectors.toList() );
+		if ( entries.isEmpty() ) {
+			// nothing to write - return fast.
+			return;
+		}
+		try {
+			moduleName.ifPresent( name -> tryToWriteLine( "== ", name, writer ) );
```

Comment:
It would be safer to also generate a stable anchor, preferably based on the artifact ID, because the name may change (it's mostly cosmetic).

If it's easier to include just the artifact ID and not the name, that's fine by me.

Just in case, this is what I mean by "anchor":

```
[[configuration-properties-aggregated-hibernate-search-engine]]
== Hibernate Search Engine
```